src/main/java/com/surrealdb/refactor/driver/WsPlaintextConnection.java

```diff
@@ -80,8 +80,21 @@ public QueryBlockResult query(String query, List<Param> params) {
                                 }
                                 JsonElement outerResultJson = resp.get("result");
                                 QueryResult[] processedOuterResults;
-                                if (outerResultJson.isJsonArray()) {
-                                    JsonArray outerResultArray = outerResultJson.getAsJsonArray();
+
+                                // checks to see if result is a JSON element, if not throw exception
+                                if (outerResultJson.isJsonObject()) {
+                                    JsonArray outerResultArray;
+
+                                    if (!outerResultJson.isJsonArray()) {
+
+                                        // add element to an array if it is not an array
+                                        outerResultArray = new JsonArray();
+                                        outerResultArray.add(outerResultJson);
+                                    } else {
```

Comment:
okay, I am working on it now. My question would be when would the else portion of the code be run?

Reply:
It's a great question! The logical answer is on primitives, and null values. But the "engineering answer" is we don't know. If the `else` handles array types, and it isn't an array type, then the error we will get won't be correct. I would much rather we have clear errors where assumptions (in this case, the type) have been broken. It will make diagnosis and debugging easier. :)